game/scripts/vscripts/abilities/oaa_zuus_cloud.lua

```diff
@@ -0,0 +1,114 @@
+zuus_cloud = class( AbilityBaseClass )
+LinkLuaModifier( "modifier_zuus_cloud_oaa", "abilities/oaa_zuus_cloud.lua", LUA_MODIFIER_MOTION_NONE )
+
+function zuus_cloud:OnSpellStart()
+  local caster = self:GetCaster()
+  local hCloud = CreateUnitByName( "npc_dota_zeus_cloud", self:GetCursorPosition(), true, caster, caster, caster:GetTeamNumber() )
+  hCloud:SetOwner( self:GetCaster() )
+  hCloud:SetControllableByPlayer( self:GetCaster():GetPlayerOwnerID(), false )
+  hCloud:AddNewModifier( caster, self, "modifier_zuus_cloud_oaa", { Interval = self:GetSpecialValueFor( "cloud_bolt_interval" ), Radius = self:GetSpecialValueFor( "cloud_radius" ) } )
+  hCloud:AddNewModifier( caster, self, "modifier_kill", { duration = self:GetSpecialValueFor( "cloud_duration" ) } )
+  FindClearSpaceForUnit( hCloud, self:GetCursorPosition(), true )
+end
+
+modifier_zuus_cloud_oaa = class( ModifierBaseClass )
+
+function modifier_zuus_cloud_oaa:IsHidden()
+  return true
+end
+
+function modifier_zuus_cloud_oaa:IsDebuff()
+  return false
+end
+
+function modifier_zuus_cloud_oaa:IsPurgable()
+  return false
+end
+
+function modifier_zuus_cloud_oaa:RemoveOnDeath()
+  return true
+end
+
+function modifier_zuus_cloud_oaa:OnCreated( kv )
+  if IsServer() then
+    self.Interval = kv.Interval
+    self.Radius = kv.Radius
+    self.cloud_particle = ParticleManager:CreateParticle("particles/units/heroes/hero_zeus/zeus_cloud.vpcf", PATTACH_ABSORIGIN, self:GetParent())
+    ParticleManager:SetParticleControl(self.cloud_particle, 1, Vector(self.Radius, 1, 1))
+
+    ParticleManager:SetParticleControlEnt(self.cloud_particle, 2, self:GetParent(), PATTACH_POINT_FOLLOW, "attach_hitloc", self:GetParent():GetAbsOrigin(), true)
+    self:StartIntervalThink( 0.1 )
+    EmitSoundOn("Hero_Zuus.Cloud.Cast", self:GetParent())
+  end
+end
+
+function modifier_zuus_cloud_oaa:OnDestroy()
+  if IsServer() then
+    ParticleManager:DestroyParticle( self.cloud_particle, false)
```

Comment:
I would ask that either everything uses spaces between the brackets or doesn't, but that's probably too nitpicky. I would like it to at least be symmetrical in a single function call though.